specifyweb/specify/views.py

```diff
@@ -391,3 +392,84 @@ def set_admin_status(request, userid):
     else:
         user.clear_admin()
         return http.HttpResponse('false', content_type='text/plain')
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Replace a new agent for an old agent.",
+            "content": {
+                "application/x-www-form-urlencoded": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The error.",
+                        "properties": {
+                            "old_agent_id": {
+                                "type": "integer",
+                                "description": "The old AgentID value of the agent record that is to be replaced by the new one."
+                            },
+                            "new_agent_id": {
+                                "type": "integer",
+                                "description": "The new AgentID value of the agent that is replacing the old one."
+                            }
+                        },
+                        'required': ['old_agent_id', 'new_agent_id'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "204": {"description": "Success",},
+            "400": {"description": "The AgentID specified does not exist."}
+        }
+    },
+})
+@login_maybe_required
+@require_POST
+def agent_record_replacement(request: http.HttpRequest, old_agent_id, new_agent_id: int) -> http.HttpResponse:
+    """Replaces all the foreign keys referencing the old AgentID
+    with the new AgentID, and deletes the old agent record.
+    """
+    check_permission_targets(request.specify_collection.id, request.specify_user.id, [])
```

Comment:
Need to integrate this endpoint with the permission system (take a look at other endpoints to see how that works).
Check for Agent Delete permission. On top of that, could probably add a special permission for this endpoint (as this endpoint modifies many tables, it would be considered a bit sensitive)

Reply:
Looks to be fixed in [d686310](https://github.com/specify/specify7/pull/2469/commits/d686310e857a224a3a62745ca9df836381829a84)